src/Persistence/PaYaml/Serialization/PaYamlSerializer.cs

```diff
@@ -128,5 +128,45 @@ private static void WriteTextWriter<TValue>(TextWriter writer, in TValue? value,
         // See: fluentvalidation.net
         // See: https://www.youtube.com/watch?v=jblRYDMTtvg
     }
+
+    /// <summary>
+    /// checks the input source is a sequence of YAML items.
+    /// </summary>
+    /// <param name="yaml">The YAML text to check.</param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    public static bool CheckIsSequence(string yaml)
+    {
+        _ = yaml ?? throw new ArgumentNullException(nameof(yaml));
+
+        using var reader = new StringReader(yaml);
+        return CheckIsSequence(reader);
+    }
+
+    /// <summary>
+    /// checks the input source is a sequence of YAML items.
+    /// </summary>
+    /// <param name="reader"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    public static bool CheckIsSequence(StringReader reader)
+    {
+        ArgumentNullException.ThrowIfNull(reader);
+
+        var builder = new DeserializerBuilder()
+            .IgnoreUnmatchedProperties()
+            .WithTypeConverter(new YamlSequenceTesterConverter());
+        var deserializer = builder.Build();
+        try
+        {
+            var results = deserializer.Deserialize<object[]>(reader);
+            return results is not null;
+        }
+        catch (YamlException)
```

Comment:
we should not need to use exceptions as control-flow.

Another design option would be to deseralize into a custom type, register a converter for that type, and in the converter populate a property which has the 'result' of the detection.

e..g
```cs
private record SequenceDetectorResult
{
  public required bool IsSequence { get; init; }
}
```